Packages/io.chainsafe.web3-unity/Editor/ServerSettings.cs

```diff
@@ -1,46 +1,119 @@
 using UnityEditor;
 using UnityEngine;
 using System;
+using System.Collections.Generic;
 using System.Threading.Tasks;
 using UnityEngine.Networking;
 using Newtonsoft.Json;
-using ChainSafe.GamingSdk.Editor;
 using System.IO;
+using System.Linq;
 using System.Text;
 using ChainSafe.Gaming.UnityPackage;
+using Microsoft.IdentityModel.Tokens;
+using UnityEditor.Experimental.GraphView;
+using UnityEngine.Events;
+using ChainInfo = ChainSafe.Gaming.UnityPackage.Model;
 
 /// <summary>
 /// Allows the developer to alter chain configuration via GUI
 /// </summary>
 public class ChainSafeServerSettings : EditorWindow
 {
+    #region Fields
+    
+    // Default values
     private const string ProjectIdPrompt = "Please enter your project ID";
-    private const string ChainIdPrompt = "Please enter your chain ID";
-    private const string ChainPrompt = "Please enter your chain i.e Ethereum, Binance, Cronos";
-    private const string NetworkPrompt = "Please enter your network i.e Mainnet, Testnet";
-    private const string SymbolPrompt = "Please enter your chain's native symbol i.e Eth, Cro";
-    private const string RpcPrompt = "Please enter your RPC endpoint";
-
+    private const string ChainIdDefault = "11155111";
+    private const string ChainDefault = "Ethereum";
+    private const string NetworkDefault = "Sepolia";
+    private const string SymbolDefault = "Seth";
+    private const string RpcDefault = "https://rpc.sepolia.org";
+    
+    // Chain values
     private string projectID;
     private string chainID;
     private string chain;
     private string network;
     private string symbol;
     private string rpc;
+    private string newRpc;
 
     Texture2D logo = null;
+    
+    // Search window
+    private StringListSearchProvider searchProvider;
+    private ISearchWindowProvider _searchWindowProviderImplementation;
+    public static UnityEvent onDropDownChange;
```

Comment:
can we reference this event from the search provider?

Reply:
done